bundles/org.openhab.binding.weatherunderground/src/main/java/org/openhab/binding/weatherunderground/internal/WeatherUndergroundHandlerFactory.java

```diff
@@ -82,7 +86,7 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
 
         if (thingTypeUID.equals(THING_TYPE_WEATHER)) {
-            return new WeatherUndergroundHandler(thing, localeProvider, unitProvider);
+            return new WeatherUndergroundHandler(thing, localeProvider, unitProvider, timeZoneProvider.getTimeZone());
```

Comment:
Shouldn't the timeZoneProvider be passed in its entirety to allow users to change the timezone without having to recreate the handler?

Reply:
At the first reading, I did not understand your point. But now I realize I am not sure how it behaves when a core system setting is updated.
The timezone is an openHAB system configuration setting that the user can update, using Paper UI for example. When this setting is updated, the bundle (which one exactly ?) is updated. As we have an OSGi reference to this bundle/interface in the binding thing handler factory component, what will happen ? Will the handler factory component be restarted with the updated reference ? If true, all thing handlers will be created again and my solution is ok. Passing the TimeZoneProvider object will make no change.
In case the setting change will have no impact on the binding handler factory component (no deactivate/activate), your remark will be valid. But in this case, I don't understand how OSGi components are informed of changes in referenced components.

Reply:
This is not clear for me in fact.
I should test a change of timezone and see the impact on the thing handler factory and thing handlers.

Reply:
Ok, tested now. In fact, changing the timezone setting has no direct impact on the thing handler factory or thing handler. Nothing is restarted. So my solution is not good.
If I pass the TimeZoneProvider as parameter to the thing handler, it looks like it is correctly updated in the thing handler.

Reply:
Fixed